src/main/java/us/abstracta/wiresham/Flow.java

```diff
@@ -144,9 +169,65 @@ public static Flow fromPcap(File file, String serverAddress, String filter) thro
   }
 
   public static Flow fromYml(File ymlFile) throws FileNotFoundException {
-    List<PacketStep> packets = new Yaml(buildYamlConstructor())
-        .load(new FileInputStream(ymlFile));
-    return new Flow(packets);
+    List<Flow> flows = new ArrayList<>();
+    new Yaml(buildYamlConstructor())
+        .loadAll(new FileInputStream(ymlFile))
+        .iterator()
+        .forEachRemaining(f -> flows.add((Flow) f));
+    parseFlowsFromYml(flows);
+    return flows.stream().filter(f -> f.getId() == null).findFirst().orElse(flows.get(0));
+  }
+
+  private static void parseFlowsFromYml(List<Flow> flows) {
```

Comment:
maybe a recursive logic simplifies this?

something like:

```java
return solveIncludes(flows.get(0), flows);

solveSteps(flow, flows) {
  ret
  lastPort
  for ( step : flow.steps) {
    if (step instanceOf IncludePacketStep) {
      ret.addAll(solveInclude(includeStep, flows))
    } else {
      if step.port != null {
        lastPort = step.port
      } else {
        step.port = lastPort
      }
      ret.add(step)
    }
  }
  return ret
}

solveInclude(includeStep, flows) {
  solveSteps(flows.stream()
    .filter(includeStep.id.equals(it.id))
    .findFirst()
    .orElseGet(() -> flows.get(Integer.parseInt(id))), flows)
}
```

Reply:
I did not kind about that when implementing it. But know makes a lot of sense to use recursion over here.
Thanks for this satisfying contribution!!